src/Http/ServerRequest.php

```diff
@@ -1410,12 +1422,26 @@ public function query($name = null)
     /**
      * Read a specific query value or dotted path.
      *
-     * @param string $name The name or dotted path to the query param.
-     * @param mixed $default The default value if the named parameter is not set.
+     * Developers are encouraged to use getQueryParams() when possible as it is PSR-7 compliant, and this method
+     * is not. You can future proof your code by using the following example.
```

Comment:
@ADmad okay, I have no idea how all these PSR-7 interfaces improve Cake. If any of the comments I added to the code are invalid. Please let me know, and I'll change it.